Update tiny-keccak from 1.5.0 to 2.0.2

iancoleman opened the pull request:
In sn_api the majority of dependencies use tiny-keccak 2.0.2 so it makes sense to have them all at the same version.

```
sn_api (master) $  cargo tree -i tiny-keccak:1.5.0
tiny-keccak v1.5.0
├── self_encryption v0.19.8
│   └── sn_client v0.46.2
│       └── sn_api v0.18.0 (/tmp/sn_api/sn_api)
│           ├── sn_authd v0.1.0 (/tmp/sn_api/sn_authd)
│           ├── sn_cli v0.18.0 (/tmp/sn_api/sn_cli)
│           └── sn_cmd_test_utilities v1.0.0 (/tmp/sn_api/sn_cmd_test_utilities)
│               [dev-dependencies]
│               └── sn_cli v0.18.0 (/tmp/sn_api/sn_cli)
├── sn_api v0.18.0 (/tmp/sn_api/sn_api) (*)
├── sn_client v0.46.2 (*)
├── sn_cmd_test_utilities v1.0.0 (/tmp/sn_api/sn_cmd_test_utilities) (*)
└── sn_data_types v0.14.3
    ├── sn_api v0.18.0 (/tmp/sn_api/sn_api) (*)
    ├── sn_client v0.46.2 (*)
    ├── sn_cmd_test_utilities v1.0.0 (/tmp/sn_api/sn_cmd_test_utilities) (*)
    ├── sn_messaging v3.0.0
    │   └── sn_client v0.46.2 (*)
    └── sn_transfers v0.3.1
        └── sn_client v0.46.2 (*)
    [dev-dependencies]
    └── sn_cli v0.18.0 (/tmp/sn_api/sn_cli)
[dev-dependencies]
└── sn_cli v0.18.0 (/tmp/sn_api/sn_cli)

sn_api (master) $  cargo tree -i tiny-keccak:2.0.2
tiny-keccak v2.0.2
├── sn_messaging v3.0.0
│   └── sn_client v0.46.2
│       └── sn_api v0.18.0 (/tmp/sn_api/sn_api)
│           ├── sn_authd v0.1.0 (/tmp/sn_api/sn_authd)
│           ├── sn_cli v0.18.0 (/tmp/sn_api/sn_cli)
│           └── sn_cmd_test_utilities v1.0.0 (/tmp/sn_api/sn_cmd_test_utilities)
│               [dev-dependencies]
│               └── sn_cli v0.18.0 (/tmp/sn_api/sn_cli)
├── threshold_crypto v0.4.0
│   ├── sn_api v0.18.0 (/tmp/sn_api/sn_api) (*)
│   ├── sn_client v0.46.2 (*)
│   ├── sn_data_types v0.14.3
│   │   ├── sn_api v0.18.0 (/tmp/sn_api/sn_api) (*)
│   │   ├── sn_client v0.46.2 (*)
│   │   ├── sn_cmd_test_utilities v1.0.0 (/tmp/sn_api/sn_cmd_test_utilities) (*)
│   │   ├── sn_messaging v3.0.0 (*)
│   │   └── sn_transfers v0.3.1
│   │       └── sn_client v0.46.2 (*)
│   │   [dev-dependencies]
│   │   └── sn_cli v0.18.0 (/tmp/sn_api/sn_cli)
│   ├── sn_messaging v3.0.0 (*)
│   └── sn_transfers v0.3.1 (*)
└── xor_name v1.1.9
    ├── sn_api v0.18.0 (/tmp/sn_api/sn_api) (*)
    ├── sn_cli v0.18.0 (/tmp/sn_api/sn_cli)
    ├── sn_client v0.46.2 (*)
    ├── sn_data_types v0.14.3 (*)
    ├── sn_messaging v3.0.0 (*)
    └── sn_transfers v0.3.1 (*)
    [dev-dependencies]
    └── sn_cli v0.18.0 (/tmp/sn_api/sn_cli)
```

bochaco reviewed (commented):
@iancoleman thanks a lot! Just clippy has some complaints. Also, can you please force push the commit using https://www.conventionalcommits.org/en/v1.0.0-beta.3/, we are following that for all our repos, so please beware for all upcoming PRs.

iancoleman commented:
Sorry I had an old version of clippy which missed some stuff. Just updated and fixed and should now be passing.

Thanks for the tips on commit message style @bochaco, I'll follow that from now on :slightly_smiling_face:
